flash/text/classification/data.py

```diff
@@ -13,138 +13,63 @@
 # limitations under the License.
 import os
 from functools import partial
-from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from typing import Any, Callable, Dict, List, Mapping, Optional, Sequence, Tuple, Union
 
 from datasets import DatasetDict, load_dataset
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
 from torch import Tensor
 from transformers import AutoTokenizer, default_data_collator
 from transformers.modeling_outputs import SequenceClassifierOutput
 
-from flash.core.classification import ClassificationState
 from flash.data.auto_dataset import AutoDataset
+from flash.data.callback import BaseDataFetcher
 from flash.data.data_module import DataModule
+from flash.data.data_source import DataSource, DefaultDataSources, LabelsState
 from flash.data.process import Postprocess, Preprocess
 
 
-class TextClassificationPreprocess(Preprocess):
-
-    def __init__(
-        self,
-        input: str,
-        backbone: str,
-        max_length: int,
-        target: str,
-        filetype: str,
-        train_file: Optional[str] = None,
-        label_to_class_mapping: Optional[Dict[str, int]] = None,
-    ):
-        """
-        This class contains the preprocessing logic for text classification
-
-        Args:
-            # tokenizer: Hugging Face Tokenizer.  # TODO: Add back a tokenizer argument and make backbone optional?
-            input: The field storing the text to be classified.
-            max_length:  Maximum number of tokens within a single sentence.
-            target: The field storing the class id of the associated text.
-            filetype: .csv or .json format type.
-            label_to_class_mapping: Dictionary mapping target labels to class indexes.
-        """
+class TextDataSource(DataSource):
 
+    def __init__(self, tokenizer, max_length: int = 128):
         super().__init__()
 
-        if label_to_class_mapping is None:
-            if train_file is not None:
-                label_to_class_mapping = self.get_label_to_class_mapping(train_file, target, filetype)
-            else:
-                raise MisconfigurationException(
-                    "Either ``label_to_class_mapping`` or ``train_file`` needs to be provided"
-                )
-
-        self.backbone = backbone
-        self.tokenizer = AutoTokenizer.from_pretrained(backbone, use_fast=True)
-        self.input = input
-        self.filetype = filetype
+        self.tokenizer = tokenizer
```

Comment:
Better to create the tokenizer in the DataSource. It is simpler to pickle.

Reply:
Done :smiley: